### DISCUSSION TITLE
Multiple document operations at once

### DISCUSSION THREAD

**Discussion by FabioArmonia** [fattureincloud/api#366] [Q&A] [answered]
Is there a way to handle multiple documents at once with one access token?
I am using the typescript-sdk for NodeJS and have tried creating two "credit notes" issued documents using javascript "Promise.all" and the server responded with the error message "Creazione/Modifica di un in corso. Riprova".

In alternative, how fast can I send data management requests to the server? I am thinking about creating a pipeline that slow requests down but i am open to other alternatives.

**Comment by valmoz** [comment, marked as answer]
Hi @FabioArmonia 
unfortunately, the issued documents are complex data structures that require a certain amount of time to be created completely.
Also, our APIs are protected by a [rate limit](https://developers.fattureincloud.it/docs/basics/limits-and-quotas#%EF%B8%8F-short-term-rate-limits) to avoid receiving too many requests in a short amount of time.

I think the solution could be one of the following (or a mix of those):
- as you suggested, adopting a queue or a pipeline to slow down the requests
- (simple, but prone to error) add a sleep between two requests
- adopt [exponential backoff](https://developers.fattureincloud.it/docs/basics/limits-and-quotas#%EF%B8%8F-keep-calm-and-deal-with-quotas) to retry with growing intervals
